checkout.js

```diff
@@ -1,5 +1,8 @@
 function selectSaleItems(items) {
-  return items
+  return items.filter(function discounted(items) {
```

Comment:
No harm in naming this function but remember it is being passed in directly and never usable anywhere else so an anonymous function is fine and is actually best practice.

Also, we name functions what they do so something like checkDiscount or filterOnDiscount would be a better name if we did need one.